### PR TITLE
fix: enable 'open in new tab' functionality when clicking link with ctrl/cmd

### LINKED ISSUE #85: Cmd + Click (Mac) and Ctrl + Click (Windows) to open a new tab
Hey, cool library !

With `gatsby-link`, cmd + click opens a link in a new tab. This same behavior does not seem to be present in `gatsby-plugin-transition-link`.

Is it possible for us to still enable this behavior without breaking the inner workings of the library?

### PR DESCRIPTION
**Pull request by pcooney10**
Fixes #85 

This was built with some inspiration from @reach/router:
 - https://github.com/reach/router/blob/master/src/index.js#L542-L545
 - https://github.com/reach/router/blob/master/src/index.js#L391-L397

### CODE REVIEW

File: src/components/TransitionLink.js [hunk 1]
```diff
@@ -31,20 +32,21 @@ const TransitionLink = ({
...
+                trigger,
+                replace,
+                linkState: state,
+                ...context
+              });
+            };
             if (typeof onClick === "function") {
               onClick(event);
-            }
+            };
```
Comment (pcooney10) [inline src/components/TransitionLink.js]: I wasn't entirely sure what to do here:
```js
if (typeof onClick === "function") {
  onClick(event);
};
```
I am assuming this is just forwarding a custom onClick event? I decided to leave it outside of the context of `shouldNavigate()`, but it maybe it needs to be in there as well.

**Comment by TylerBarnes** [comment]
Hey @pcooney10 thanks for this! This is something I hadn't thought of and it looks like it's working great.

File: src/components/TransitionLink.js (hunk 1 above)
Comment (TylerBarnes) [reply src/components/TransitionLink.js]: I agree that this should be outside of shouldNavigate. If an engineer wants their onClick to not fire on cmd+click, they can add a check for that in their onClick function.

Comment (TylerBarnes) [reply src/components/TransitionLink.js]: Actually, just pushed a commit where we pass down wether navigation is happening or not to the onClick, that way the engineer implementing this doesn't need to do the same check we're already doing.

**Comment by TylerBarnes** [comment]
Thanks again @pcooney10 , upgrade to `gatsby-plugin-transition-link@1.12.4` to use your contribution!

**Pull request merged**